include/cadmium/modeling/dynamic_model.hpp

```diff
@@ -0,0 +1,128 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CADMIUM_ATOMIC_HPP
+#define CADMIUM_ATOMIC_HPP
+
+#include <cadmium/modeling/dynamic_message_bag.hpp>
+
+namespace cadmium {
+    namespace dynamic {
+        namespace modeling {
+
+            struct EOC {
+                std::string _from;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                EOC() = delete;
+
+                EOC(std::string from, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _from(from), _link(l) {}
+
+                EOC(const EOC& other)
+                        : _from(other._from), _link(other._link) {}
+            };
+
+            struct EIC {
+                std::string _to;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                EIC() = delete;
+
+                EIC(std::string to, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _to(to), _link(l) {}
+
+                EIC(const EIC& other)
+                        : _to(other._to), _link(other._link) {}
+            };
+
+            struct IC {
+                std::string _to;
+                std::string _from;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                IC() = delete;
+
+                IC(std::string to, std::string from, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _to(to), _from(from), _link(l) {}
+
+                IC(const IC& other)
+                        : _to(other._to), _from(other._from), _link(other._link) {}
+            };
+
+            using dynamic_EC=std::tuple<std::type_index, std::type_index, std::type_index>;
```

Comment:
Is this missing a letter or you keeping same type for all External couplings? (dynamic_EC) 
If it is the second it worth a comment mentioning.

Reply:
It was originally the same type for all External Couplings, but it is now a deprecated types, so I have removed this statement and the "using dynamic_IC=..." statement from the code and no comment is needed.